appGeo/src/app/game/game.component.ts

```diff
@@ -1,15 +1,142 @@
 import { Component, OnInit } from '@angular/core';
+import {Player, Killer, Civilian} from '../player/player.component';
+import{CampusMap} from '../map/campus-map.component'
+//A CampusMap is a Map of the Bubbles that exist on campus
+
+const INACTIVE = 0
+const ACTIVE = 1
+const CIVILIAN = 0
+const KILLER = 1
 
 @Component({
   selector: 'ns-game',
   templateUrl: './game.component.html',
   styleUrls: ['./game.component.css']
 })
-export class GameComponent implements OnInit {
+export class Game implements OnInit {
+  gameActive: number //a boolean number
+  currentTime: Date //a Date object
+  endTime: Date //a Date object
+
+  map: CampusMap //a map object
+
+  players: Map<number, Player> //a hashmap of mapping playerID ints to players
+  snapshots: Map<number, Snapshot> //a hashmap of mapping snapshotID ints to snapshots
+  chats: Map<number, Chat> //a hashmap of mapping chatsID ints to chats
+
 
-  constructor() { }
+  constructor(endTime: Date, gameMap: CampusMap, players: Map<number, Player>) {
+    this.endTime = endTime
+        this.map = gameMap
+
+        if(players != undefined) {
+            this.players = players
+        } else {
+            this.players = new Map()
+        }
+
+        this.snapshots = new Map()
+        this.chats = new Map()
+  }
 
   ngOnInit(): void {
   }
 
+  startGame() {
+    this.#setGameActive(ACTIVE)
+
+    //Randomly generate killer objects for 20% of the players and civilians for the rest
+    //Rebuild the players list with these objects
+    //TODO
+  }
+
+  endGame() {
+      this.#setGameActive(INACTIVE)
+
+      //TODO
+  }
+
+  getGameActive() {
+      return this.gameActive
+  }
+
+  #setGameActive(status) {
+      this.gameActive = status
+  }
+
+  getCurrentTime() {
+      return this.currentTime
+  }
+
+  getEndTime() {
+      return this.endTime
+  }
+
+  setEndTime(endTime) {
+      this.endTime = endTime
+  }
+
+  getPlayer(playerID) {
+      return this.players.get(playerID)
+  }
+
+  addPlayer(player) {
+      this.players.set(player.getUserID(), player)
+  }
+
+  removePlayer(playerID) {
+    return this.players.delete(playerID)
+  }
+
+  get PlayerCount() {
+    return this.players.size
+  }
+
+  getRoleCount(countKiller) {
+
+      const playersList = this.map[Symbol.iterator]();//need to adjust the code for Campus Map for this to work
+
+
+      var count = 0
+
+      for(var p in playersList) {
+          if(!(typeof p == Killer) ^ (countKiller)) {
```

Comment:
This was poorly written by me originally. But should be swapped to
`if((playersList[i] instanceof Killer) && (countKiller) || 
               !(playersList[i] instanceof Killer) && !(countKiller))`
to fit in typescript

Reply:
I adjusted it according to your suggestion. Hope it looks as intended now